lightpack.py

```diff
@@ -204,12 +348,119 @@ def getCountLeds(self, fresh=True):
+		:type fresh: boolean
+		:returns: integer
+		"""
+		if fresh or self._maxLeds is None:
+			self._maxLeds = int(self._sendAndReceivePayload('getmaxleds'))
+		return self._maxLeds
+
+	def _ledDimensionRead(self, command):
+		"""
+		Read a LED size state snippet into rectangle Tuple.
+
+		:param command: LED size state snipped
+		:type command: str
+		:returns: Tuple with LED and tuple of x0, y0, x1, y1.
+		"""
+		try:
+			parts = command.split('-', 1)
+			coordinates = parts[1].split(',', 3)
+			rectangle = [int(x) for x in coordinates if x.strip()]
+			led = parts[0].strip()
+			if not led or len(rectangle) != 4:
+				return None
+			return int(led), tuple(rectangle)
+		except IndexError:
+			return None
+		except ValueError:
+			return None
+
+	def getLedSizes(self, fresh=True):
+		"""
+		Get the dimensions of all LEDs.
+
+		If the parameter fresh (default True) is set to False, a previously 
+		cached value will be used if available.
+
+		:param fresh: fetch a new value or use cached response
+		:type fresh: boolean
+		:returns: Dictionary of tuples of x-position, y-position, height and
+		width, using 0-based LED numbers as keys
+		"""
+		if fresh or self._ledSizes == {}:
+			commands = self._sendAndReceivePayload('getleds').rstrip(';') \
+				.split(';')
+			self._ledSizes = {}
+			for command in commands:
+				data = self._ledDimensionRead(command)
+				if data is not None:
+					self._ledSizes[data[0]] = data[1]
+		return self._ledSizes
+
+	def getCountMonitors(self, fresh=True):
+		"""
+		Get the number of monitors the Lightpack controls.
+
+		If the parameter fresh (default True) is set to False, a previously 
+		cached value will be used if available.
+
+		:param fresh: fetch a new value or use cached response
+		:type fresh: boolean
+		:returns: integer
+		"""
+		if fresh or self._countMonitors is None:
+			self._countMonitors = int(self._sendAndReceivePayload('countmonitors'))
+		return self._countMonitors
+
+	def getMonitorSize(self, monitor, fresh=True):
+		"""
+		Get the dimensions of a monitor.
+
+		If the parameter fresh (default True) is set to False, a previously 
+		cached value will be used if available.
+
+		:param monitor: 0-based monitor index
+		:type monitor: integer
+		:param fresh: fetch a new value or use cached response
+		:type fresh: boolean
+		:returns: tuple of x-position, y-position, width and height
+		"""
+		if fresh or self._monitor == {} or monitor not in self._monitor:
+			try:
+				response = self._sendAndReceivePayload('getsizemonitor:%s' %
+													   monitor)
+				coordinates = response.split(',', 3)
+				rectangle = [int(x) for x in coordinates if x.strip()]
+				self._monitor[monitor] = tuple(rectangle)
+			except AttributeError:
+				return None
+		return self._monitor[monitor]
+
+	def getLockStatus(self):
+		"""
+		Get the API lock status (locked or unlocked).
+
+		:returns: string, 'ok' or 'no' depending on lock state.
+		"""
+		return self._sendAndReceivePayload('getlockstatus')
```

Comment:
Seems an odd choice on the API side to have the responses `ok` and `no`... Might make sense to normalize these to a boolean?

Reply:
True, how about renaming it `isLocked()` and returning a boolean?

Reply:
That sounds better to me. Unless that would break away from method naming closely following the API calls themselves?

Reply:
Actually, there is also a "busy" state possible. Bummer. 🙁 I'll adjust the description at least.

https://github.com/Atarity/Lightpack/blob/4dee73a443cba4c4073291febe450e6c1941f3af/Software/src/ApiServer.cpp#L554